src/components/Card/Card.tsx

```diff
@@ -1,16 +1,54 @@
 import './Card.scss';
 import { Movie } from '../../types';
+import { useAppSelector } from '../../hooks/redux';
+import { ChangeEvent, useEffect, useState } from 'react';
+import { moviesApi } from '../../services/apiService';
 
 interface CardProps {
   movie: Movie;
 }
 
-export default function Card({ movie }: CardProps) {
+export default function Card({ movie: { id, title, year, poster } }: CardProps) {
+  const posterUrl = poster === 'N/A' ? './poster.svg' : poster;
+  const posterClass = poster === 'N/A' ? 'placeholder' : 'poster';
+
+  const selectedMovies = useAppSelector((state) => state.movies.selectedMovies);
+  const isSelected = selectedMovies.some((selectedMovie) => selectedMovie.id === id);
+  const [checked, setChecked] = useState(isSelected);
+
+  const [fetchMovie] = moviesApi.useLazyFetchMovieByIdQuery();
+
+  useEffect(() => {
+    setChecked(isSelected);
+  }, [isSelected]);
+
+  const handleCheckboxChange = (e: ChangeEvent<HTMLInputElement>) => {
+    e.stopPropagation();
+    const newChecked = !checked;
+    setChecked(newChecked);
+    fetchMovie(id);
```

Comment:
This handler sends request on every click, despite switched was checkbox checked, or not. You can add a condition to avoid that. We don't need an useless requests.